Update snowflake-setup.md (#4680)
fishtown-analytics/docs.getdbt.com: website/docs/docs/core/connect-data-platform/snowflake-setup.md

### Key Pair Authentication

> To use key pair authentication, omit a `password` and instead provide a `private_key_path` and, optionally, a `private_key_passphrase`. 

Reviewer: ```suggestion
To use key pair authentication, skip the `password` and provide a `private_key_path`. If needed, you can also add a `private_key_passphrase`. 
```

> **Note:** Versions of dbt before 0.16.0 required that private keys were encrypted and a `private_key_passphrase` was provided. Since dbt 0.16.0, unencrypted private keys are allowed. Only add the passphrase if necessary.

Reviewer: ```suggestion
**Note**: In dbt versions before 0.16.0, private keys needed encryption and a `private_key_passphrase`. From dbt version 0.16.0 onwards, unencrypted private keys are accepted, so add a passphrase only if necessary.
```

> **Note**: In dbt versions before 0.16.0, private keys needed encryption and a `private_key_passphrase`. From dbt version 0.16.0 onwards, unencrypted private keys are accepted, so add a passphrase only if necessary.

Reviewer 2: Removing the "prior to .16.0" language since this is long since deprecated

```suggestion
**Note**: Unencrypted private keys are accepted, so add a passphrase only if necessary.
```
